scheduler/setup_funcs.go

```diff
@@ -53,3 +56,21 @@ func cacheSimilarFailing(comparator *CmpBasedTaskComparator) (err error) {
 	}
 	return nil
 }
+
+// groupTaskGroups puts tasks that have the same build and task group next to
+// each other in the queue. This ensures that, in a stable sort,
+// byTaskGroupOrder sorts task group members relative to each other.
+func groupTaskGroups(comparator *CmpBasedTaskComparator) error {
+	taskMap := make(map[string]task.Task)
+	taskKeys := []string{}
+	for _, t := range comparator.tasks {
+		k := fmt.Sprintf("%s_%s_%s", t.BuildId, t.TaskGroup, t.Id)
```

Comment:
if I asked really nicely could you use hyphens or dots rather than underscores?

Reply:
lol sure